crates/aptos-faucet/core/src/endpoints/fund.rs

```diff
@@ -281,16 +283,32 @@ impl FundApiComponents {
         // Same thing, this uses FromRequest.
         header_map: &HeaderMap,
         dry_run: bool,
+        asset: Option<String>,
     ) -> poem::Result<Vec<SignedTransaction>, AptosTapError> {
+        // Add logging for the asset parameter
+        println!("fund_inner called with asset: {:?}", asset);
+
         let (checker_data, bypass, _semaphore_permit) = self
             .preprocess_request(&fund_request, source_ip, header_map, dry_run)
             .await?;
 
-        // Fund the account.
-        let fund_result = self
-            .funder
-            .fund(fund_request.amount, checker_data.receiver, false, bypass)
-            .await;
+        // Fund the account - handle asset parameter
+        let fund_result = match &asset {
+            Some(asset_name) => {
+                // NEW: Multi-asset path
+                info!("Using multi-asset flow for asset: {}", asset_name);
+                self.funder
+                    .fund(fund_request.amount, checker_data.receiver, Some(asset_name.clone()), false, bypass)
```

Comment:
Seems like it'd be cleaner to resolve the asset into a definite non optional value before this. Right now there are 3 variants of calls to fund:
 
1. ShelbyUSD
1. APT explicit
1. APT implicit

It'd be better to just have two.